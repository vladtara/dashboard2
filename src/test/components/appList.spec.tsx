import { render } from "@testing-library/react";
import AppList, { IAppListProps } from "../../components/appList";

const props: IAppListProps = {
  categories: [
    {
      name: "Category Test",
      items: [
        {
          name: "App Test",
          icon: "bug_report",
          url: "#",
          displayURL: "test",
          newTab: false,
        },
        {
          name: "App Test",
          icon: "bug_report",
          url: "#",
          displayURL: "test",
          newTab: false,
        },
      ],
    },
  ],
  apps: [
    {
      name: "App Test",
      icon: "bug_report",
      url: "#",
      displayURL: "test",
      newTab: false,
    },
  ],
};

describe("appList.tsx", () => {
  it("Tests AppList rendering with categories and apps", () => {
    const { baseElement } = render(
      <AppList categories={props.categories} apps={props.apps} />,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("Tests AppList rendering with categories", () => {
    const { baseElement } = render(<AppList categories={props.categories} />);
    expect(baseElement).toMatchSnapshot();
  });

  it("Tests AppList rendering with apps", () => {
    const { baseElement } = render(<AppList apps={props.apps} />);
    expect(baseElement).toMatchSnapshot();
  });

  it("Tests AppList rendering with neither", () => {
    const { baseElement } = render(<AppList />);
    expect(baseElement).toMatchSnapshot();
  });
});
