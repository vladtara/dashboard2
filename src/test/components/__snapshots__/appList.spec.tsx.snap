// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appList.tsx Tests AppList rendering with apps 1`] = `
<body>
  <div>
    <div
      class="sc-iBkjds ixywls"
    >
      <h2
        class="sc-ftvSup bdxGxw"
      >
        Applications
      </h2>
      <div
        class="sc-evZas csYxrp"
      >
        <ul
          class="sc-jqUVSM egPCCj"
        >
          <li
            class="sc-kDDrLX OpKuo"
          >
            <a
              class="sc-dkzDqf gHJogt"
              href="#"
            >
              <div
                class="sc-hKMtZM kjGwCV"
              >
                <i
                  class="sc-gsnTZi ctUcHQ"
                >
                  bug_report
                </i>
              </div>
              <div
                class="sc-eCYdqJ bcsQqv"
              >
                <div
                  class="sc-jSMfEi bkhPVp"
                >
                  App Test
                </div>
                <p
                  class="sc-gKXOVf EDUWD"
                >
                  test
                </p>
              </div>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</body>
`;

exports[`appList.tsx Tests AppList rendering with categories 1`] = `
<body>
  <div>
    <div
      class="sc-iBkjds ixywls"
    >
      <h2
        class="sc-ftvSup bdxGxw"
      >
        Applications
      </h2>
      <div
        class="sc-evZas csYxrp"
      >
        <h3
          class="sc-papXJ sc-crXcEl lhcacf ieLYyf"
        >
          Category Test
        </h3>
        <ul
          class="sc-jqUVSM egPCCj"
        >
          <li
            class="sc-kDDrLX OpKuo"
          >
            <a
              class="sc-dkzDqf gHJogt"
              href="#"
            >
              <div
                class="sc-hKMtZM kjGwCV"
              >
                <i
                  class="sc-gsnTZi ctUcHQ"
                >
                  bug_report
                </i>
              </div>
              <div
                class="sc-eCYdqJ bcsQqv"
              >
                <div
                  class="sc-jSMfEi bkhPVp"
                >
                  App Test
                </div>
                <p
                  class="sc-gKXOVf EDUWD"
                >
                  test
                </p>
              </div>
            </a>
          </li>
          <li
            class="sc-kDDrLX OpKuo"
          >
            <a
              class="sc-dkzDqf gHJogt"
              href="#"
            >
              <div
                class="sc-hKMtZM kjGwCV"
              >
                <i
                  class="sc-gsnTZi ctUcHQ"
                >
                  bug_report
                </i>
              </div>
              <div
                class="sc-eCYdqJ bcsQqv"
              >
                <div
                  class="sc-jSMfEi bkhPVp"
                >
                  App Test
                </div>
                <p
                  class="sc-gKXOVf EDUWD"
                >
                  test
                </p>
              </div>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</body>
`;

exports[`appList.tsx Tests AppList rendering with categories and apps 1`] = `
<body>
  <div>
    <div
      class="sc-iBkjds ixywls"
    >
      <h2
        class="sc-ftvSup bdxGxw"
      >
        Applications
      </h2>
      <div
        class="sc-evZas csYxrp"
      >
        <h3
          class="sc-papXJ sc-crXcEl lhcacf ieLYyf"
        >
          Category Test
        </h3>
        <ul
          class="sc-jqUVSM egPCCj"
        >
          <li
            class="sc-kDDrLX OpKuo"
          >
            <a
              class="sc-dkzDqf gHJogt"
              href="#"
            >
              <div
                class="sc-hKMtZM kjGwCV"
              >
                <i
                  class="sc-gsnTZi ctUcHQ"
                >
                  bug_report
                </i>
              </div>
              <div
                class="sc-eCYdqJ bcsQqv"
              >
                <div
                  class="sc-jSMfEi bkhPVp"
                >
                  App Test
                </div>
                <p
                  class="sc-gKXOVf EDUWD"
                >
                  test
                </p>
              </div>
            </a>
          </li>
          <li
            class="sc-kDDrLX OpKuo"
          >
            <a
              class="sc-dkzDqf gHJogt"
              href="#"
            >
              <div
                class="sc-hKMtZM kjGwCV"
              >
                <i
                  class="sc-gsnTZi ctUcHQ"
                >
                  bug_report
                </i>
              </div>
              <div
                class="sc-eCYdqJ bcsQqv"
              >
                <div
                  class="sc-jSMfEi bkhPVp"
                >
                  App Test
                </div>
                <p
                  class="sc-gKXOVf EDUWD"
                >
                  test
                </p>
              </div>
            </a>
          </li>
        </ul>
      </div>
      <div
        class="sc-evZas csYxrp"
      >
        <h3
          class="sc-papXJ sc-crXcEl lhcacf ieLYyf"
        >
          Uncategorized apps
        </h3>
        <ul
          class="sc-jqUVSM egPCCj"
        >
          <li
            class="sc-kDDrLX OpKuo"
          >
            <a
              class="sc-dkzDqf gHJogt"
              href="#"
            >
              <div
                class="sc-hKMtZM kjGwCV"
              >
                <i
                  class="sc-gsnTZi ctUcHQ"
                >
                  bug_report
                </i>
              </div>
              <div
                class="sc-eCYdqJ bcsQqv"
              >
                <div
                  class="sc-jSMfEi bkhPVp"
                >
                  App Test
                </div>
                <p
                  class="sc-gKXOVf EDUWD"
                >
                  test
                </p>
              </div>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</body>
`;

exports[`appList.tsx Tests AppList rendering with neither 1`] = `
<body>
  <div />
</body>
`;
